File: eco-core/core-nms/v1_17_R1/src/main/java/com/willfp/eco/proxy/v1_17_R1/FastItemStackFactory.java
package com.willfp.eco.proxy.v1_17_R1;

import com.willfp.eco.core.fast.FastItemStack;
import com.willfp.eco.proxy.FastItemStackFactoryProxy;
import com.willfp.eco.proxy.v1_17_R1.fast.EcoFastItemStack;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public final class FastItemStackFactory implements FastItemStackFactoryProxy {
    @Override
    public FastItemStack create(@NotNull final ItemStack itemStack) {
        return new EcoFastItemStack(itemStack);
    }
}

File: eco-core/core-nms/v1_17_R1/src/main/kotlin/com/willfp/eco/proxy/v1_17_R1/FastItemStackFactory.kt
package com.willfp.eco.proxy.v1_17_R1

import com.willfp.eco.core.fast.FastItemStack
import com.willfp.eco.proxy.v1_17_R1.fast.NMSFastItemStack
import org.bukkit.inventory.ItemStack
import com.willfp.eco.proxy.FastItemStackFactoryProxy

class FastItemStackFactory : FastItemStackFactoryProxy {
    override fun create(itemStack: ItemStack): FastItemStack {
        return NMSFastItemStack(itemStack)
    }
}